### PR TITLE
[Merged by Bors] - feat: port Algebra.GroupWithZero.Basic

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
Tracking mathlib commit: [39af7d3bf61a98e928812dbc3e16f4ea8b795ca3](https://github.com/leanprover-community/mathlib/commit/39af7d3bf61a98e928812dbc3e16f4ea8b795ca3)

### CODE REVIEW

**Comment by Ruben-VandeVelde** [comment]
I somehow made the `librarySearch` test angry, but I'm not sure how.

**Comment by mcdoll** [comment]
Can you please please claim your ports over at https://github.com/leanprover-community/mathlib/wiki/mathlib4-port-status

**Comment by pechersky** [comment]
> Can you please please claim your ports over at https://github.com/leanprover-community/mathlib/wiki/mathlib4-port-status

I've added it

**Comment by semorrison** [comment]
The Lean 4 fix should arrive in `nightly-2022-11-24`.

File: Mathlib/Algebra/GroupWithZero/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+theorem inv_mul_cancel_left₀ (h : a ≠ 0) (b : G₀) : a⁻¹ * (a * b) = b :=
+  calc
+    a⁻¹ * (a * b) = a⁻¹ * a * b := (mul_assoc _ _ _).symm
+    _ = b := by simp [h]
+
+
+private theorem inv_eq_of_mul (h : a * b = 1) : a⁻¹ = b := by
+  rw [← inv_mul_cancel_left₀ (left_ne_zero_of_mul_eq_one h) b, h, mul_one]
+--#align doesn't seem to work with private theorems
+--#align inv_eq_of_mul inv_eq_of_mul
```
Comment (semorrison) [inline Mathlib/Algebra/GroupWithZero/Basic.lean:251]:
```suggestion
```

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**